### ISSUE TITLE
Unable to load govcms profile

### ISSUE THREAD

**Issue by marblegravy** [govCMS/govcms7-scaffold#12]
Using the master or develop branches at the moment, I can't get this to install the govCMS profile. Not sure where it's supposed to find it, but it is definitely having trouble:

The environment is fine, drush is installed OK, but the govcms profile doesn't want to play.

```
[govcms7-scaffold]cli-drupal:/app$ drush si govcms
You are about to DROP all tables in your 'drupal' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the --notify global option.                                           [ok]
Exception: Sorry, the profile you have chosen cannot be loaded. in /app/docroot/includes/install.core.inc:1384                      [error]
Stack trace:
#0 /app/docroot/includes/install.core.inc(471): install_load_profile(Array)
#1 /app/docroot/includes/install.core.inc(339): install_run_task(Array, Array)
#2 /app/docroot/includes/install.core.inc(77): install_run_tasks(Array)
#3 /app/vendor/drush/drush/includes/drush.inc(727): install_drupal(Array)
#4 /app/vendor/drush/drush/includes/drush.inc(713): drush_call_user_func_array('install_drupal', Array)
#5 /app/vendor/drush/drush/commands/core/drupal/site_install_7.inc(86): drush_op('install_drupal', Array)
#6 /app/vendor/drush/drush/commands/core/site_install.drush.inc(271): drush_core_site_install_version('govcms', Array)
#7 /app/vendor/drush/drush/includes/command.inc(422): drush_core_site_install('govcms')
#8 /app/vendor/drush/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#9 /app/vendor/drush/drush/includes/command.inc(199): drush_command('govcms')
#10 /app/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#11 /app/vendor/drush/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#12 /app/vendor/drush/drush/drush.php(12): drush_main()
#13 {main}
```

Any tips on what's missing?

**Comment by fubarhouse** [comment]
@marblegravy ,

I've just done a test of this, here's my output.

```
➜  ~ cd test
➜  test git clone https://github.com/govCMS/govcms7-scaffold.git
Cloning into 'govcms7-scaffold'...
remote: Enumerating objects: 53, done.
remote: Counting objects: 100% (53/53), done.
remote: Compressing objects: 100% (38/38), done.
remote: Total 395 (delta 21), reused 39 (delta 13), pack-reused 342
Receiving objects: 100% (395/395), 343.52 KiB | 545.00 KiB/s, done.
Resolving deltas: 100% (211/211), done.
➜  test cd govcms7-scaffold
➜  govcms7-scaffold git:(develop) ahoy up
WARNING: The LAGOON_ENVIRONMENT_TYPE variable is not set. Defaulting to a blank string.
Creating network "govcms7-scaffold_default" with the default driver
Building test
Step 1/6 : FROM govcmslagoon/test
 ---> da688aaaa31c
Step 2/6 : COPY themes/ /app/sites/default/themes/custom
 ---> bbc823ecadb1
Step 3/6 : RUN rm -Rf /app/tests/behat/features/*
 ---> Running in 10a4df7de863
Removing intermediate container 10a4df7de863
 ---> d3f6af1daaca
Step 4/6 : COPY tests/behat/features/ /app/tests/behat/features
 ---> 8875e57ca95c
Step 5/6 : RUN rm -Rf /app/tests/phpunit/tests/*
 ---> Running in eaa074ac90a9
Removing intermediate container eaa074ac90a9
 ---> 7a8789ae8d0c
Step 6/6 : COPY tests/phpunit/tests/ /app/tests/phpunit/tests
 ---> 959578bc72ed

Successfully built 959578bc72ed
Successfully tagged govcms7-scaffold_test:latest
WARNING: Image for service test was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Building nginx
Step 1/4 : ARG CLI_IMAGE
Step 2/4 : FROM ${CLI_IMAGE} as cli
 ---> d316c7009f89

Step 3/4 : FROM govcmslagoon/nginx-drupal
 ---> 355ecf62993c
Step 4/4 : COPY --from=cli /app /app
 ---> Using cache
 ---> a95ee84c66ae

Successfully built a95ee84c66ae
Successfully tagged govcms7-scaffold_nginx:latest
WARNING: Image for service nginx was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Building php
Step 1/4 : ARG CLI_IMAGE
Step 2/4 : FROM ${CLI_IMAGE} as cli
 ---> d316c7009f89

Step 3/4 : FROM govcmslagoon/php
 ---> eb773405a012
Step 4/4 : COPY --from=cli /app /app
 ---> Using cache
 ---> ed8249714c59

Successfully built ed8249714c59
Successfully tagged govcms7-scaffold_php:latest
WARNING: Image for service php was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating govcms7-scaffold_mariadb_1 ... done
Creating govcms7-scaffold_cli_1     ... done
Creating govcms7-scaffold_php_1     ... done
Creating govcms7-scaffold_nginx_1   ... done
Creating govcms7-scaffold_test_1    ... done
Creating govcms7-scaffold_chrome_1  ... done
➜  govcms7-scaffold git:(develop) ahoy install
WARNING: The LAGOON_ENVIRONMENT_TYPE variable is not set. Defaulting to a blank string.
You are about to DROP all tables in your 'drupal' database. Do you want to continue? (y/n): y
Starting Drupal installation. This takes a while. Consider using the        [ok]
--notify global option.
Installation complete.  User name: admin  User password: YY7fkgEQjn         [ok]
Thank you for installing the Metatag module. It is recommended to       [status]
read the module's README.txt file as there are some known issues that
may affect this site.
It may be worth verifying on the Settings page which types of content   [status]
on the site should allow meta tags.
To use menu blocks, find the "Add menu block" link on the administer    [status]
blocks page.
Password policy module successfully installed. Please review the        [status]
available configuration settings.
Risky permissions have been automatically disabled.                     [status]
To use Superfish go to the administer blocks page and enable a          [status]
Superfish block.
Metatag support has been enabled for the Tags vocabulary.               [status]
Metatag support has been enabled for the Slide content type.            [status]
Metatag support has been enabled for the Standard page content type.    [status]
Metatag support has been enabled for the Blog Article content type.     [status]
Metatag support has been enabled for the Event content type.            [status]
Metatag support has been enabled for the Publication content type.      [status]
Metatag support has been enabled for the Media Release content type.    [status]
Metatag support has been enabled for the News Article content type.     [status]
Metatag support has been enabled for the Promotion content type.        [status]
Content permissions have been rebuilt.                                  [status]
The sitemap links were rebuilt.                                         [status]
Content permissions have been rebuilt.                                  [status]
Login Security and Flood Control modules are automatically disabled.   [warning]
WARNING: The LAGOON_ENVIRONMENT_TYPE variable is not set. Defaulting to a blank string.
Database dump saved to                                                 [success]
/app/sites/default/files/private/backups/pre-deploy-dump.sql.gz
No database updates required                                           [success]
'all' cache was cleared.                                               [success]
The following extensions will be enabled: govcms_lagoon, lagoon_logs, clamav, fast_404, redis
Do you really want to continue? (y/n): y
clamav was enabled successfully.                                            [ok]
govcms_lagoon was enabled successfully.                                     [ok]
redis was enabled successfully.                                             [ok]
fast_404 was enabled successfully.                                          [ok]
lagoon_logs was enabled successfully.                                       [ok]
The following extensions will be disabled: govcms_lagoon
Do you really want to continue? (y/n): y
govcms_lagoon was disabled successfully.                                    [ok]
The following modules will be uninstalled: govcms_lagoon
Do you really want to continue? (y/n): y
govcms_lagoon was successfully uninstalled.                                 [ok]
➜  govcms7-scaffold git:(develop)
```

What I think is happening given some offline context would be that your base images have been tailored to your specific needs. If I'm not mistaken, you can circumvent this temporarily by running `ahoy build` or `docker-compose build`. When returning to your custom implementation of this scaffold, you would run into similar issues and by running this command again would restore the local images to the expected state.

This scenario would be easy to test if you can identify if any custom modules (which you manage) are included in the build you described above.

**Comment by tobybellwood** [comment]
if it can't find the profile - is it actually in the container?
```
tobyb@TOBY_PC:/c/Users/tobyb/sites/docker/govcms7-scaffold$ docker-compose exec cli bash
WARNING: The LAGOON_ENVIRONMENT_TYPE variable is not set. Defaulting to a blank string.
[govcms7-scaffold]cli-drupal:/app$ find / -name govcms
/app/profiles/govcms
/app/profiles/govcms/themes/govcms
[govcms7-scaffold]cli-drupal:/app$ ls -al /app/profiles/govcms
total 172
drwxr-xr-x 1 root root  4096 Sep  3 23:46 .
drwxr-xr-x 1 root root  4096 Sep  3 23:43 ..
-rw-r--r-- 1 root root  2058 Sep  3 23:43 .ahoy.yml
drwxr-xr-x 2 root root  4096 Sep  3 23:43 .circleci
drwxr-xr-x 2 root root  4096 Sep  3 23:43 .docker
-rw-r--r-- 1 root root   194 Sep  3 23:43 .dockerignore
-rw-r--r-- 1 root root    39 Sep  3 23:43 .editorconfig
-rw-r--r-- 1 root root   854 Sep  3 23:43 .env.default
-rw-r--r-- 1 root root   643 Sep  3 23:43 .gitattributes
-rw-r--r-- 1 root root   960 Sep  3 23:43 .gitignore
-rw-r--r-- 1 root root  4546 Sep  3 23:43 README.md
-rw-r--r-- 1 root root   361 Sep  3 23:43 RELEASE.md
drwxr-xr-x 3 root root  4096 Sep  3 23:43 build
-rw-r--r-- 1 root root   373 Sep  3 23:43 build-govcms.make
drwxr-xr-x 2 root root  4096 Sep  3 23:43 css
-rw-r--r-- 1 root root  2352 Sep  3 23:43 docker-compose.yml
-rw-r--r-- 1 root root   272 Sep  3 23:43 drupal-org-core.make
-rw-r--r-- 1 root root 12195 Sep  3 23:43 drupal-org.make
-rw-r--r-- 1 root root  4825 Sep  3 23:45 govcms.info
-rw-r--r-- 1 root root 48217 Sep  3 23:43 govcms.install
-rw-r--r-- 1 root root  2978 Sep  3 23:43 govcms.profile
drwxr-xr-x 2 root root  4096 Sep  3 23:43 images
drwxr-xr-x 2 root root  4096 Sep  3 23:43 js
drwxr-xr-x 1 root root  4096 Sep  3 23:45 libraries
drwxr-xr-x 1 root root  4096 Sep  3 23:44 modules
-rw-r--r-- 1 root root  3625 Sep  3 23:43 phpcs.xml
drwxr-xr-x 5 root root  4096 Sep  3 23:43 tests
drwxr-xr-x 1 root root  4096 Sep  3 23:45 themes
[govcms7-scaffold]cli-drupal:/app$   
```

**Comment by marblegravy** [comment]
@fubarhouse You're on the money - ```ahoy build``` reset it to what I expected the govcms-scaffold should look like.

Following @tobybellwood 's line of thinking I was able to confirm that my build was pulling in one of our development images for our paas sites. 

which is why govcms wasn't available.
```[govcms7-scaffold]cli-drupal:/app$ find / -name govcms``` returned no result.

After running ```ahoy build``` however, it is behaving as I had expected it to:

```
[govcms7-scaffold]cli-drupal:/app$ find / -name govcms
/app/profiles/govcms
/app/profiles/govcms/themes/govcms
```

I've submitted a pull request for your consideration to avoid this confusion if anyone else came across this situation.
